### PR TITLE
On-chain view documentation page

### LINKED ISSUE #1117: Abstraction to execute contract views
**Is your feature request related to a problem and use case? Please describe.**
We want to provide a new feature on the ContractAbstraction class that will allow users to execute the new views proposed in the Hangzhou protocol.

**Describe the solution you'd like**
The user creates an instance of ContractAbstraction using the `at` method of the Contract or Wallet API. The `contractViews` member of the `ContractAbstraction` instance is dynamically populated with methods that match the on-chain view names. _`contractViews` is an object where the key is the view name, and the value is a function that returns an instance of the OnChainView class._
The view parameter is expected in the JS object format and not as flattened arguments.
_Note for reference, the flattened arguments are the expected format when calling a contract entrypoint using the methods member, but we plan to move away from this format and favor the object one (see the methodsObject vs methods members of the ContractAbstraction)._
A method named `getSignature` on the `OnChainView` class allows inspecting the parameter and the returned type of the view. The `executeView` method of the `OnChainView` class allows simulating the view. It takes as a parameter a viewCaller which represents the contract address which is the caller of the view and an optional source which is the public key hash of the account that initialized this view execution.

Here is an example where a contract script contains an on-chain view called fib:
```
const contract = await Tezos.contract.at('KT1...');
contract.contractViews.fib().getSignature();
// getSignature returns: { parameter: 'nat', result: 'nat' }
await contract.contractViews.fib(10).executeView({ viewCaller: 'KT1... });
// executeView returns the BigNumber 55
```

The implementation to execute the view can be very similar to the Michelson view from the tzip16 package.

**Note: on-chain view should not be confused with the tzip4 views which are represented by ContractAbstraction.views.**

### PR DESCRIPTION
**Pull request by roxaneletourneau**
Closes #1117 
Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [ ] Your code builds cleanly without any errors or warnings
- [ ] You have added unit tests
- [ ] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

File: docs/on_chain_views.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+       VIEW "fib" nat;
+       IF_SOME {NIL operation ; PAIR;} { FAIL }
+     }
+```
+
+**Example of calling a contract entrypoint that makes a call to a view using Taquito:**
+
+The following live code example shows a contract (`contractCallFib`) calling the view `fib` of another contract (`contractTopLevelViews`). 
+
+The example first shows the initial storage of the contract `contractCallFib`. It calls the default entry point of `contractCallFib` with the value of its storage + 1 and the address of the contract `contractTopLevelViews`. A call is made the `fib` view of `contractTopLevelViews` with the `storage + 1` as argument. The view returns the value of the Fibonacci sequence at the position represented by `storage + 1`. The storage of `contractCallFib` is updated to the result of the view.
```
Comment (michaelkernaghan) [inline docs/on_chain_views.md:112]: "A call is made the "  ->  "A call is made to the .."

File: docs/on_chain_views.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+  })
+  .catch((error) => println(`Error: ${JSON.stringify(error, null, 2)}`));
+```
+
+## How to simulate a view execution using Taquito
+
+Taquito offers the ability to simulate the result of on-chain views. 
+
+The user can create an instance of `ContractAbstraction` using the `at` method of the Contract or Wallet API with the contract's address that defines the views. The `contractViews` member of the `ContractAbstraction` instance is dynamically populated with methods that match the on-chain view names. 
+
```
Comment (michaelkernaghan) [inline docs/on_chain_views.md:146]: " view arguments in parameter and "...    

should it be "the parameter",  ?

**Review by michaelkernaghan — COMMENTED** [review]
very nice. just one, maybe two very minor typos.
There is an article about the views,  "https://medium.com/@keefertaylor/introducing-views-in-the-harbinger-price-oracle-aeee793e4577" in which they say views help to avoid the need to use callbacks in contracts. Could we mention something about that? or will readers already know about it?

File: docs/on_chain_views.md (hunk 1 above)
Comment (roxaneletourneau) [reply docs/on_chain_views.md:112]: Fixed, thanks!

File: docs/on_chain_views.md (hunk 2 above)
Comment (roxaneletourneau) [reply docs/on_chain_views.md:146]: I think my sentence is not really clear :P

It is better:
*`contractViews` is an object where the key is the view name, and the value is a function that takes the view arguments as a parameter and returns an instance of `OnChainView` class.*

**Comment by roxaneletourneau** [comment]
> views help to avoid the need to use callbacks in contracts

thanks, @michaelkernaghan, I added a note about it.

**Pull request merged**